File: Common/Domain/PerAssemblySequencePointsCoverage.fs
namespace R4nd0mApps.TddStud10.Common.Domain

open System.Collections.Concurrent
open System

[<Serializable>]
type PerAssemblySequencePointsCoverage = 
    inherit DataStoreEntityBase<AssemblyId, ConcurrentBag<SequencePointCoverage>>
    
    new() = 
        { inherit DataStoreEntityBase<_, _>() }
        then ()
    
    member public t.Serialize path = DataStoreEntityExtensions.Serialize<PerAssemblySequencePointsCoverage> path t

    static member public Deserialize path = DataStoreEntityExtensions.Deserialize<PerAssemblySequencePointsCoverage> (path)

File: TestHost/PerAssemblySequencePointsCoverage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;
using R4nd0mApps.TddStud10.Common.Domain;

namespace R4nd0mApps.TddStud10.Engine
{
    [Serializable]
    public class PerAssemblySequencePointsCoverage : SerializableDictionary<AssemblyId, List<SequencePointCoverage>>
    {
        public static PerAssemblySequencePointsCoverage Deserialize(string file)
        {
            return Deserialize<PerAssemblySequencePointsCoverage>(file);
        }
    }
}
